tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -467,22 +468,44 @@ public function adminHasAddedUserToGroupUsingTheGraphApi(
 	}
 
 	/**
-	 * @When /^the administrator creates a group "([^"]*)" using the Graph API$/
 	 *
 	 * @param string $group
+	 * @param ?string $user
 	 *
-	 * @return void
-	 * @throws Exception
+	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function adminCreatesGroupUsingTheGraphApi(string $group): void {
-		$response = GraphHelper::createGroup(
+	public function createGroup(string $group, ?string $user = null): ResponseInterface {
+		if ($user) {
+			$username = $user;
+			$password = $this->featureContext->getPasswordForUser($user);
+		} else {
+			$username = $this->featureContext->getAdminUsername();
+			$password = $this->featureContext->getAdminPassword();
+		}
+		return GraphHelper::createGroup(
 			$this->featureContext->getBaseUrl(),
 			$this->featureContext->getStepLineRef(),
-			$this->featureContext->getAdminUsername(),
-			$this->featureContext->getAdminPassword(),
+			$username,
+			$password,
 			$group,
 		);
+	}
+
+	/**
+	 * @When /^the administrator creates a group "([^"]*)" using the Graph API$/
+	 * @When user :user creates a group :group using the Graph API
+	 * @When user :user tries to create a group :group using the Graph API
```

Comment:
```suggestion
	 * @When /^(the administrator|user "([^"]*)") (tries to )?create(s)? a group "([^"]*)" using the Graph API$/
```
regex could be improved to catch all cases

Reply:
or even
```php
/^(the administrator|user "([^"]*)") (tries to create|creates) a group "([^"]*)" using the Graph API$/
```

Reply:
Hmmm, could be done by a single regex. :thinking: But won't it be  a bit difficult to read (determine which steps are using the function below)
IDE can easily navigate in either of the cases, I guess. What should I do? @kiranparajuli589

Reply:
Yes,,valid point ;) would surely minimize the readability. anything u prefer @saw-jan